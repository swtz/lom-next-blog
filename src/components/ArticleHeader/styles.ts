import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      font-size: ${theme.font.sizes.xxlarge};
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.lteMedium} {
        font-size: ${theme.font.sizes.large};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    margin: ${theme.spacings.medium} 0;

    @media ${theme.media.lteMedium} {
      font-size: 2rem;
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
