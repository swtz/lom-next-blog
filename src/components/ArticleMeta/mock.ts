import { ArticleMetaProps } from '.';

export default {
  createdAt: '2025-02-12T11:41:13.991Z', // value from current Strapi API
  author: {
    data: {
      id: '1',
      attributes: {
        displayName: 'surfista',
        slug: 'surfista',
      },
    },
  },

  categories: {
    data: [
      {
        id: '4',
        attributes: {
          displayName: 'Off Topic',
          slug: 'off-topic',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'tech',
          slug: 'tech',
        },
      },
    ],
  },
} as ArticleMetaProps;
