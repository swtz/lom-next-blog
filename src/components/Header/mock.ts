import { HeaderProps } from '.';

export default {
  blogName: 'Blog do Arroz Ã© vida!',
  blogDescription: `Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa.`,
  logo: '/assets/images/logo.svg',
  showText: true,
} as HeaderProps;
