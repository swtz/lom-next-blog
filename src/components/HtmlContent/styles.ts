import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: calc(${theme.font.sizes.xsmall} + 0.2rem);
    line-height: 1.5;

    p {
      margin: ${theme.spacings.medium} 0;
    }

    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }

    code {
      font-family: monospace;
      color: ${theme.colors.secondary};
      font-size: ${theme.font.sizes.xsmall};
      background: ${theme.colors.mediumGray};
      padding: 0.2rem;
      margin: 0.2rem;
    }

    pre {
      background: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacings.medium} 0;
      font-size: ${theme.font.sizes.xsmall};
      width: 100%;
      overflow-x: auto;
    }

    pre code {
      color: inherit;
      background: inherit;
    }

    img {
      max-width: 100%;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.medium} 0;
    }

    .image figcaption {
      background: ${theme.colors.mediumGray};
      font-size: ${theme.font.sizes.xsmall};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }

    .image-style-side {
      float: right;
      max-width: 50%;
      margin: ${theme.spacings.medium} ${theme.spacings.small};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }

    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }

    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }

    .media iframe {
      width: 100%;
      height: auto;
      aspect-ratio: 16 / 9;
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.darkerGray};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      pre {
        font-size: 1.2rem;
      }

      .image-style-side {
        float: none;
        max-width: 100%;
        margin: ${theme.spacings.xlarge} 0;
      }
    }
  `};
`;
