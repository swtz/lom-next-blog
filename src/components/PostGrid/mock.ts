import { data } from '../../api/dados.json';

export const posts = data.posts;
