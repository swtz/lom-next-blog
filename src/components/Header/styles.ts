import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    /* padding: ${theme.spacings.xhuge}; */
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.xsmall};
    margin: 0 auto;

    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacings.xsmall} - 1rem);
    }

    @media ${theme.media.lteXSmall} {
      flex-flow: column wrap;

      > div {
        ${HeadingStyles} {
          margin: ${theme.spacings.xsmall} 0
            calc(${theme.spacings.small} - 1rem);
        }
      }
    }
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`;
