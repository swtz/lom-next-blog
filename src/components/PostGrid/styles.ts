import styled, { css } from 'styled-components';
import { Wrapper as PostCardStyles } from '../PostCard/styles';

export const Wrapper = styled.div`
  ${() => css``}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    justify-content: center;
    align-items: center;
    grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
    gap: 3rem;
    padding: 1rem;

    ${PostCardStyles} {
      padding: 1rem;
      border: 0.5rem solid ${theme.colors.secondary};
      border-radius: 5%;
      height: 100%;
      transition: border-color 300ms ease-in-out;
    }

    ${PostCardStyles}:hover {
      border-color: ${theme.colors.darkText};
    }
  `}
`;
