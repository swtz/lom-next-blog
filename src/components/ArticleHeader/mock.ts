import { ArticleHeaderProps } from '.';
import { ArticleMetaProps } from '../ArticleMeta';

const metadata = {
  createdAt: '2025-02-12T11:41:13.991Z',
  author: {
    data: {
      id: '1',
      attributes: {
        displayName: 'surfista',
        slug: 'surfista',
      },
    },
  },

  categories: {
    data: [
      {
        id: '4',
        attributes: {
          displayName: 'Off Topic',
          slug: 'off-topic',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'tech',
          slug: 'tech',
        },
      },
    ],
  },
} as ArticleMetaProps;

export default {
  title: 'Hello, rice! With rice is possible make a several delicious dishes',
  excerpt: `Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa (Asian rice).`,
  cover: {
    data: {
      id: 1,
      attributes: {
        url: '/assets/images/arroz-branco-saudavel.webp',
        alternativeText: null,
      },
    },
  },
  metadata: metadata,
} as ArticleHeaderProps;
