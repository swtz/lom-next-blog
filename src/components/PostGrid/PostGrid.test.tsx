import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostGrid, PostGridProps } from '.';
import { posts } from './mock';

const props: PostGridProps = { posts };

describe('<PostGrid />', () => {
  it('should render the correct number of PostCard components', () => {
    renderTheme(<PostGrid {...props} />);

    expect(screen.getAllByRole('heading')).toHaveLength(
      props.posts.data.length,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<PostGrid {...props} />);

    expect(container).toMatchSnapshot();
  });
});
