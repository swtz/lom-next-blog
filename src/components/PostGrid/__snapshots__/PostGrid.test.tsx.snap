// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostGrid /> should match snapshot 1`] = `
.c6 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c3 {
  text-align: center;
}

.c3 .c5 {
  margin: 0;
  margin-top: .8rem;
  margin-bottom: .8rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c3 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover .c5 {
  color: #DC143C;
}

.c3:hover img {
  opacity: 0.8;
}

.c4 {
  max-width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c0 {
  width: 100%;
  max-width: 120rem;
  margin: 0 auto;
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: repeat(auto-fill,minmax(300px,1fr));
  gap: 3rem;
  padding: 1rem;
}

.c1 .c2 {
  padding: 1rem;
  border: 0.5rem solid #DC143C;
  border-radius: 5%;
  height: 100%;
  -webkit-transition: border-color 300ms ease-in-out;
  transition: border-color 300ms ease-in-out;
}

.c1 .c2:hover {
  border-color: #333333;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
      >
        <a
          href="/post/hello-rice"
        >
          <img
            alt="a fox holding a box containing a bottle and ice cubes"
            class="c4"
            src="https://pub-aea127bc483c40dea7b9339692296909.r2.dev/stages_of_rice_milling_8cb950424c.jpeg"
          />
        </a>
        <a
          href="/post/hello-rice"
        >
          <h2
            class="c5 c6"
          >
            Hello, rice!
          </h2>
        </a>
        <p
          class=""
        >
          Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa (Asian rice).
        </p>
      </div>
      <div
        class="c2 c3"
      >
        <a
          href="/post/hello-fox"
        >
          <img
            alt="a fox holding a box containing a bottle and ice cubes"
            class="c4"
            src="https://pub-aea127bc483c40dea7b9339692296909.r2.dev/DALL_E_2024_10_26_12_58_20_A_modern_logo_design_for_Deck_da_Praia_Conveniencia_featuring_a_fox_mascot_representing_cold_beer_delivery_The_fox_should_appear_in_motion_with_sp_fd834176b6.webp"
          />
        </a>
        <a
          href="/post/hello-fox"
        >
          <h2
            class="c5 c6"
          >
            Hello, fox!
          </h2>
        </a>
        <p
          class=""
        >
          Foxes are small-to-medium-sized omnivorous mammals belonging to several genera of the family Canidae.
        </p>
      </div>
      <div
        class="c2 c3"
      >
        <a
          href="/post/hello-bean"
        >
          <img
            alt="a fox holding a box containing a bottle and ice cubes"
            class="c4"
            src="https://pub-aea127bc483c40dea7b9339692296909.r2.dev/how_to_cook_dried_beans_6fb0921283.jpg"
          />
        </a>
        <a
          href="/post/hello-bean"
        >
          <h2
            class="c5 c6"
          >
            Hello, bean!
          </h2>
        </a>
        <p
          class=""
        >
          A bean is the seed of any plant in the legume family (Fabaceae) used as a vegetable for human consumption or animal feed.
        </p>
      </div>
    </div>
  </div>
</div>
`;
